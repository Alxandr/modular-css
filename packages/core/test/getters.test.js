"use strict";

const namer = require("test-utils/namer.js");
const { find, cwd, relative } = require("test-utils/fixtures.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("getters", () => {
        let processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer,
                cwd,
            });
        });
        
        describe(".file", () => {
            it("should return all the files that have been added", async () => {
                await processor.file(find("start.css"));
                await processor.file(find("local.css"));

                const files = Object.keys(processor.files);
                
                expect(files.map(relative)).toMatchSnapshot();
            });
        });

        describe(".options", () => {
            it("should return the merged options object", () => {
                expect(typeof processor.options).toBe("object");
            });
        });
    });
});
