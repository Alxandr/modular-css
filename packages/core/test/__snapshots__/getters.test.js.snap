// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js getters .file should return all the files that have been added 1`] = `
Array [
  "start.css",
  "local.css",
  "folder/folder.css",
]
`;
